src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/NumberHandlingTests.cs

```diff
@@ -1508,39 +1508,30 @@ public class ClassWith_NumberHandlingOn_ObjectProperty
         [ActiveIssue("Need to tweak number handling option registration following code-gen support.")]
         public static void Attribute_NotAllowed_On_Property_WithCustomConverter()
         {
-            string json = @"";
-            InvalidOperationException ex = Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize<ClassWith_NumberHandlingOn_Property_WithCustomConverter>(json));
-            string exAsStr = ex.ToString();
-            Assert.Contains(typeof(ConverterForInt32).ToString(), exAsStr);
-            Assert.Contains(typeof(ClassWith_NumberHandlingOn_Property_WithCustomConverter).ToString(), exAsStr);
+            string json = @"{""Prop"":1}";
 
-            ex = Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(new ClassWith_NumberHandlingOn_Property_WithCustomConverter()));
-            exAsStr = ex.ToString();
-            Assert.Contains(typeof(ConverterForInt32).ToString(), exAsStr);
-            Assert.Contains(typeof(ClassWith_NumberHandlingOn_Property_WithCustomConverter).ToString(), exAsStr);
+            // Ensure custom converter is honored.
+            var obj = JsonSerializer.Deserialize<ClassWith_NumberHandlingOn_Property_WithCustomConverter>(json);
+            Assert.Equal(25, obj.Prop);
+            Assert.Throws<NotImplementedException>(() => JsonSerializer.Serialize(obj));
```

Comment:
I was initially confused by this assertion, but then I realized it's an exception thrown by the custom converter. Do we need to test for this here? Or we do, shouldn't we be testing an implemented converter?

Reply:
The read side is implemented to return 25, so I believe it is implied the write side will of course get called and thus is kept simple by just throwing instead of writing 25.

There are tests later that has both read- and write- sides